pkg/controller/awssqs/reconcile.go

```diff
@@ -0,0 +1,194 @@
+	if err != nil {
+		src.Status.MarkNoSink("NotFound", "")
+		return src, err
+	}
+	src.Status.MarkSink(sinkURI)
+
+	_, err = r.createReceiveAdapter(ctx, src, sinkURI)
+	if err != nil {
+		logger.Error("Unable to create the receive adapter", zap.Error(err))
+	}
+	src.Status.MarkDeployed()
+
+	return src, nil
+}
+
+func (r *reconciler) createReceiveAdapter(ctx context.Context, src *v1alpha1.AwsSqsSource, sinkURI string) (*v1.Deployment, error) {
+	ra, err := r.getReceiveAdapter(ctx, src)
+	if err != nil && !apierrors.IsNotFound(err) {
+		logging.FromContext(ctx).Error("Unable to get an existing receive adapter", zap.Error(err))
+		return nil, err
+	}
+
+	if ra != nil {
+		logging.FromContext(ctx).Desugar().Info("Reusing exising receive adapter", zap.Any("receiveAdapter", ra))
+		return ra, nil
+	}
+
+	svc := resources.MakeReceiveAdapter(&resources.ReceiveAdapterArgs{
+		Image:   r.receiveAdapterImage,
+		Source:  src,
+		Labels:  getLabels(src),
+		SinkURI: sinkURI,
+	})
+
+	if err := controllerutil.SetControllerReference(src, svc, r.scheme); err != nil {
+		return nil, err
+	}
+
+	err = r.client.Create(ctx, svc)
+	logging.FromContext(ctx).Desugar().Info("Receive Adapter created.", zap.Error(err), zap.Any("receiveAdapter", svc))
+	return svc, err
+}
+
+func (r *reconciler) getReceiveAdapter(ctx context.Context, src *v1alpha1.AwsSqsSource) (*v1.Deployment, error) {
+	dl := &v1.DeploymentList{}
+	err := r.client.List(ctx, &client.ListOptions{
+		Namespace:     src.Namespace,
+		LabelSelector: r.getLabelSelector(src),
+		// TODO this is only needed by the fake client. Real K8s does not need it. Remove it once
+		// the fake is fixed.
+		Raw: &metav1.ListOptions{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: v1.SchemeGroupVersion.String(),
+				Kind:       "Deployment",
+			},
+		},
+	}, dl)
+
+	if err != nil {
+		logging.FromContext(ctx).Desugar().Error("Unable to list deployments: %v", zap.Error(err))
+		return nil, err
+	}
+	for _, dep := range dl.Items {
+		if metav1.IsControlledBy(&dep, src) {
+			return &dep, nil
+		}
+	}
+	return nil, apierrors.NewNotFound(schema.GroupResource{}, "")
+}
+
+func (r *reconciler) getLabelSelector(src *v1alpha1.AwsSqsSource) labels.Selector {
+	return labels.SelectorFromSet(getLabels(src))
+}
+
+func (r *reconciler) addFinalizer(s *v1alpha1.AwsSqsSource) {
+	finalizers := sets.NewString(s.Finalizers...)
+	finalizers.Insert(finalizerName)
+	s.Finalizers = finalizers.List()
+}
+
+func (r *reconciler) removeFinalizer(s *v1alpha1.AwsSqsSource) {
+	finalizers := sets.NewString(s.Finalizers...)
+	finalizers.Delete(finalizerName)
+	s.Finalizers = finalizers.List()
+}
+
+func getLabels(src *v1alpha1.AwsSqsSource) map[string]string {
+	return map[string]string{
+		"knative-eventing-source":      controllerAgentName,
+		"knative-eventing-source-name": src.Name,
+	}
+}
+
+func generateSubName(src *v1alpha1.AwsSqsSource) string {
+	return fmt.Sprintf("knative-eventing-%s-%s-%s", src.Namespace, src.Name, src.UID)
```

Comment:
I don't know if this is possible, but if we can, we should use a separator that can't be in names or namespaces, e.g. underscore rather than dash.